### PR TITLE
Rename controller to kwok and add ci

### LINKED ISSUE #9: [kwok] Rename `controller` to `kwok` in `cmd` and `image`
/kind feature

### PR DESCRIPTION
**Pull request by wzshiming**
/kind cleanup
Fixes #9

### CODE REVIEW

**Review by yibozhuang — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -33,12 +33,16 @@ verify:
...
+	$(GO_CMD) build -o bin/kwok ./cmd/kwok/*.go
 
 .PHONY: integration-test
 integration-test:
 	@echo "Not implemented yet"
 
 .PHONY: e2e-test
-e2e-test:
-	@echo "Not implemented yet"
+e2e-test: \
```
Comment (Huang-Wei) [inline Makefile:43]: `\` is not needed, right?

File: Makefile [hunk 2]
```diff
@@ -33,12 +33,16 @@ verify:
...
+	e2e-test-kwok
+
+.PHONY: e2e-test-kwok
+e2e-test-kwokctl:
```
Comment (Huang-Wei) [inline Makefile:47]: can you make this part of `e2e-test` as there is no CI job for `e2e-test-kwok`.

File: Makefile (hunk 1 above)
Comment (wzshiming) [reply Makefile:43]:
``` Makefile
.PHONY: e2e-test
e2e-test: \
	e2e-test-kwok \
	e2e-test-kwokctl

.PHONY: e2e-test-kwok
e2e-test-kwok:
	./test/kwok/kwok.test.sh

.PHONY: e2e-test-kwokctl
e2e-test-kwokctl:
	./test/kwok/kwokctl.test.sh
```

This is the final result I expect.
`\` is for the next line to follow.

File: Makefile (hunk 2 above)
Comment (wzshiming) [reply Makefile]: Ooops, e2e didn't work before

``` console
[1](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kwok/11/pull-kwok-e2e-test-main/1557193367748612096#1:build-log.txt%3A1)
./test/kwok/kwok.test.sh
[2](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kwok/11/pull-kwok-e2e-test-main/1557193367748612096#1:build-log.txt%3A2)
./test/kwok/../../images/kwok/build.sh: line 18: docker: command not found
[3](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kwok/11/pull-kwok-e2e-test-main/1557193367748612096#1:build-log.txt%3A3)
./test/kwok/kwok.test.sh: line 24: kind: command not found
[4](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kwok/11/pull-kwok-e2e-test-main/1557193367748612096#1:build-log.txt%3A4)
./test/kwok/kwok.test.sh: line 26: kind: command not found
[5](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kwok/11/pull-kwok-e2e-test-main/1557193367748612096#1:build-log.txt%3A5)
./test/kwok/kwok.test.sh: line 28: kubectl: command not found
[6](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kwok/11/pull-kwok-e2e-test-main/1557193367748612096#1:build-log.txt%3A6)
./test/kwok/kwok.test.sh: line 28: kubectl: command not found
[7](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kwok/11/pull-kwok-e2e-test-main/1557193367748612096#1:build-log.txt%3A7)
./test/kwok/kwok.test.sh: line 33: kubectl: command not found
```

docker, kind, kubectl , none of them, I don't know what to do.

Comment (wzshiming) [reply Makefile]: This case deploying `kwok` in a `kind` cluster and checking if it works.
I ignore the e2e-test to change the name first.

Comment (Huang-Wei) [reply Makefile]: > Maybe we need to change the base image of the ci

Yes, the current image is a golang based one. Please submit a PR in test-infra to use separate image.

File: Makefile (hunk 1 above)
Comment (Huang-Wei) [reply Makefile]: Or, just use one e2e entry, and local test can be run individually by shell argument?

```
e2e-test:
	./hack/e2e-test.sh
```

By default, `hack/e2e-test.sh` runs all e2e test. And we can run kwok / kwokctl by: `hack/e2e-test.sh kwok [or kwokctl]`.

**Comment by wzshiming** [comment]
Waiting merged of kubernetes/test-infra#27100 and `/retest`

File: test/kwok/kwok.test.sh [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DIR="$(dirname "${BASH_SOURCE[0]}")"
+
+DIR="$(realpath "${DIR}")"
+
+KUBECTL_VERSION=1.24.2
```
Comment (yibozhuang) [inline test/kwok/kwok.test.sh:20]: do we want to pin the version here?

just wondering

so would this mean that when we add new k8s major version, we also need to update this?

Comment (Huang-Wei) [reply test/kwok/kwok.test.sh:20]: I think we can leave it as-is for now, and update each version along with deps bump, or work on a more smart way to autodetect the k8s version.

**Comment by Huang-Wei** [comment]
@wzshiming CI still failed after adding the dind preset.

**Comment by wzshiming** [comment]
It looks CI passed

**Comment by wzshiming** [comment]
> ~@wzshiming CI still failed after adding the dind preset.~
> 
> Nvm, we need to merge #5 first.

Waiting #5 be merged
/hold

**Pull request merged**